### ISSUE TITLE
execute/refetch should be async 

### ISSUE THREAD

**Issue by jtparrett** [simoneb/axios-hooks#51]
I would like the ability to await refetch/execute and receive the response data in return.

```
const data = await execute()
// ... use data here
```

**Comment by astone123** [comment]
I'd also like the option to do this. I want to be able to catch the error thrown by the Axios call in my form submit handler.

eg.

```
try {
  const data = await execute()
} catch (error) {
 // handle request error myself
}
```

**Comment by davidwilson3** [comment]
I was able to achieve this using the transformResponse portion of the execute function:


```
const [{ error: fileError }, onSubmit] = useAxios(
        { url: "/api/submit", method: "POST" },
        { manual: true },
    );

onSubmit({
                params: { user: user, filename: "example" },
                data: { body: exampleData },
                transformResponse: [
                    (response: string) => {               
                        fileError 
                            ? errorFunction(response) // do error stuff                             
                            :  handleResponseFromAPI(reponse); //do response stuff
                    },
                ],
```

**Comment by simoneb** [comment]
Addressed in version 1.6.0

**Comment by jtparrett** [comment]
@simoneb thank you for the quick update, you're a hero :pray:
